activesupport/lib/active_support/claims.rb

```diff
@@ -0,0 +1,63 @@
+require 'active_support/time'
+
+module ActiveSupport
+  # +Claims+ supports <tt>ActiveSupport::MessageVerifier</tt> by
+  # representation of message in the form JWT Claims and verifying
+  # the claims of the signed message. 
+  class Claims
+    class InvalidClaims < StandardError; end
+    class ExpiredClaims < StandardError; end
+    
+    attr_reader :payload, :purpose, :expires_at
+
+    # Keyword argument +:payload+ is mandatory
+    def initialize(payload:, **options)
+      @payload = payload
+      @purpose = self.class.pick_purpose(options)
+      @expires_at = pick_expiration(options)
+    end
+
+    class << self
+      attr_accessor :expires_in
+
+      def pick_purpose(options)
+        options.fetch(:for) { 'universal' }
+      end
+
+      def verify!(claims, options = {})
+        raise InvalidClaims if claims[:for] != pick_purpose(options)
+        claims[:pld] if parse_expiration(claims[:exp])
```

Comment:
```pld``` is something we introduced. ```exp``` is actually there in the JWT spec and all other parameter names are also three lettered. Here, http://self-issued.info/docs/draft-ietf-oauth-json-web-token.html. It would be odd to have ```payload``` there.